### ISSUE TITLE
Sms windows is empty

### ISSUE THREAD

**Issue by wargreen** [andyholmes/gnome-shell-extension-gsconnect#699]
**Describe the bug**

After upgrade to v28, the SMS windows still empty.

**Screenshots**

If applicable, add screenshots to help explain your problem. You can drag-and-drop or cut-and-paste images directly into this edit window, to include them in your report.

**Support Log**

Can't get any log. Disable/enable the sms plugin don't show anything in the log.

**System Details (please complete the following information):**

 - **GSConnect version:** 28
   - **Installed from:** GNOME Extensions Website
 - **GNOME/Shell version:** 3.30.2-11
 - **Distro/Release:** Debian SID

**GSConnect environment (if applicable):**

 - **Paired Device(s):** Lineage OS
 - **KDE Connect app version:** 1.13.2

**Additional Notes:**

I've try to reinstall completly GSconnet, same issue. The file sms.json is empty.

**Comment by ferdnyc** [comment]
This _sounds_ like a duplicate of #320, but I'm curious @wargreen ... 
1. You say "After upgrade to v28...", does that mean SMS was _working_ prior to the upgrade?
1. What version of GSConnect were you running before v28?
1. What's the device, Android version, and SMS app you're pairing with?
1. Also, have you tried the "Legacy SMS Support" switch at the very bottom of the device's Advanced configuration?

**Comment by andyholmes** [comment]
Can you try having the window open and clicking the `[+]` in the top right corner, while generating a support log?

**Comment by dedepapa** [comment]
I have the same problem. SMS history conversation is empty.

    GSConnect version: 28
        Installed from: GNOME Extensions Website
        Distro/Release: Ubuntu 18.04.3LTS
GSConnect environment:
    Paired Device(s): Xiaomi Redmi Note 5
    KDE Connect app version: 1.13.2
I downgraded Kdeconnect to version 1.12.7 and everything is now working perfectly. SMS history conversation is there.

**Comment by krzysdz** [comment]
@dedepapa In your case it's a problem with KDE Connect Android app and not with GSConnect. There was a bug (affecting MIUI) which was fixed yesterday in [this commit](https://invent.kde.org/kde/kdeconnect-android/commit/4a0c53610f5b8f82162bda0a2ac81d789f00e64a).

**Comment by andyholmes** [comment]
Hooray, hopefully that fixes it for other phones too. Thanks for your work on that @krzysdz!

**Comment by dedepapa** [comment]
@krzysdz Thanks for your feedback and work. I'm not an expert but I suppose we have to wait for the Kdeconnect update now.

**Comment by wargreen** [comment]
> This _sounds_ like a duplicate of #320, but I'm curious @wargreen ...
> 
>     1. You say "After upgrade to v28...", does that mean SMS was _working_ prior to the upgrade?
With 27 the SMS windows don't show up #660 but the cache seem to be filled in.
> 
>     2. What version of GSConnect were you running before v28?
v27 from the gnome extension webpage.
> 
>     3. What's the device, Android version, and SMS app you're pairing with?
I use the same phone, a Samsung Galaxy S4 Mini with lineageOS 14.1.20 (android 7.1.2) ans the build-in sms app.
> 
>     4. Also, have you tried the "Legacy SMS Support" switch at the very bottom of the device's Advanced configuration?
Yes and nothing more.
I don't know if it's related but my contacts don't appear in the SMS windows.
And no error or message in my syslog when i open the SMS windows, disconnect/connect the phone, or clean the sms cache via the GUI...

Is it possible that is the same KDEconnect issue ? I've tried to downgrade it (from f-droid) but the installation fail...

**Comment by andyholmes** [comment]
Closing as a duplicate of #320. 

The contacts issue is probably caused by a similar issue, which is usually Android manufacturers not following the sqlite database conventions used by Android's content providers. There's a [blurb in the Wiki](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Help#contacts-dont-sync) about this.

**Comment by ferdnyc** [comment]
@andyholmes Well, [Lineage](https://lineageos.org/) is a community OS, the Galaxy S4 line were actually pretty gloriously rootable, installable, etc. (I had one before my current GxS6, which is the exact opposite.)

It's still definitely _possible_ the Lineage developers messed something up, of course, especially if they took any pains to ape Samsung's stock OS build in any way. But it's not directly a Samsung system, despite the manufacturer, and it would be a bit odd for community ROM developers to be in that camp — they certainly have little _reason_ to do so, and if it is their config breaking things it may be entirely unintentional. (Not that it's any help with this issue)

It's also a bit weird that it worked with earlier versions, though.

**Comment by andyholmes** [comment]
There was one of those classic inverted-logic checks [awhile back](https://invent.kde.org/kde/kdeconnect-android/commit/ffbceb9b8e2ed0031b5d23c2cdae31ab8a577635), as a part of the original attempt to fix Samsung devices. But that's been fixed since kdeconnect-android 1.13.

There was also a [change in SMS packet protocol](https://invent.kde.org/kde/kdeconnect-android/commit/abcb6cbf332bc8c145c2338ab1bd5d882a4e327d), but again that was back in 1.13 and we still have backward compatible support for that anyways.

To be honest, it's hard to really tell what's happening if we're not provided logs, and generally unless there's an actually errors in the log it's usually not our fault, so :man_shrugging:. Without evidence to the contrary, this sounds like all the other "SMS doesn't work" bugs.

**Comment by wargreen** [comment]
The sms function have worked and i havn't do any upgrade of my lineageOS, so i don't think that is an OS problem. And it's sure that is not a Samsung issue cause i havn't any samsung or proprietary apps installed.
So maybe an KdeConnect issue ?

**Comment by andyholmes** [comment]
It's hard to tell, because you've not posted any logs of using the message window. It's really quite easy; I even in built in a function for doing it: [Help: Generate Support Log](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Help#generate-support-log).

**Comment by krzysdz** [comment]
@wargreen Install KDE Connect from source or using [apt-get](https://packages.debian.org/sid/kdeconnect) and check if you can see messages there. If the issue persists with KDE Connect, then it probably is #320 - a problem with KDE Connect Android app.

**Comment by wargreen** [comment]
> It's hard to tell, because you've not posted any logs of using the message window. It's really quite easy; I even in built in a function for doing it: [Help: Generate Support Log](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Help#generate-support-log).

This tool don't show anything and i havn't any message from gsconnect in my syslog : 
```

GSConnect Version: 28
GSConnect Install: user
GJS: 15801
XDG_SESSION_TYPE: wayland
GDMSESSION: gnome
--------------------------------------------------------------------------------
Hint: You are currently not seeing messages from other users and the system.
      Users in the 'systemd-journal' group can see all messages. Pass -q to
      turn off this notice.
No journal files were opened due to insufficient permissions.
```
I will try the KDEconnect way a bit later, thank you for the tip !

**Comment by andyholmes** [comment]
Your log is telling you why you're not getting any debug messages:

> Hint: You are currently not seeing messages from other users and the system.
>      Users in the 'systemd-journal' group can see all messages.

Probably your system is set to store the journal in RAM (aka `volatile`) so it's somewhere in `/run` that your user doesn't have access to. You'd have to either add yourself to the `systemd-journal` group or set the journal storage to `persistent`.

**Comment by andyholmes** [comment]
If you don't see lines like this when you press the `[+]` button in the message window, then you aren't getting the proper log:

```
[/service/protocol/core.js:send:329]: Nexus 4: {
  "id": 1573614370222,
  "type": "kdeconnect.contacts.request_all_uids_timestamps",
  "body": {}
}
[/service/protocol/core.js:send:329]: Nexus 4: {
  "id": 1573614370331,
  "type": "kdeconnect.sms.request_conversations",
  "body": {}
}
```
These message are sent unconditionally whenever the button is pressed to resync contacts and messages. As far as I know, if GLib is compiled with systemd support, that is the log `g_log ()` will write to, not syslog.

Of course, as described in the Wiki, the *Generate Support Log* dialog must stay open or these messages are not logged or recorded anywhere.

**Comment by wargreen** [comment]
Thank you for theses details.
When I click on the [+] button in the sms windows, i got only this log :
`
Nov 13 15:21:23 fetide org.gnome.Shell.Extensions.GSConnect[14026]: [/service/protocol/core.js:send:321]: Mercredi: {#012  "id": 1573654883085,#012  "type": "kdeconnect.sms.request_conversations",#012  "body": {}#012}
`

My contacts are not synced by gsconnect cause i use our Nextcloud for this via GOA and are shown but the Contact application. And it has worked with Gsconnect v26 if i remember correctly.

**Comment by andyholmes** [comment]
If there is no response from the phone when we send that request, then the problem is not in GSConnect. You'll have to debug the Android app and file an upstream bug report:

```sh
adb logcat --pid=$(adb shell pidof -s org.kde.kdeconnect_tp)
```

**Comment by wargreen** [comment]
Ok, it seem to the Samsung compat mode is enabled but is not the stock rom...
It's strange than it has worked by the past... I don't see how to sign up on the kde's gitlab to fill up an issue. Have you an account ?
Here the kdeconnect's log :
`11-17 19:24:00.052 25935  8421 I SMSHelper: Samsung compatibility mode enabled. This may cause some features to not work properly.
11-17 19:24:00.094 25935  8421 I SMSHelper: Samsung compatibility mode enabled. This may cause some features to not work properly.
11-17 19:24:00.172 25935  8421 E KDE/Device: Exception in SMSPlugin's onPacketReceived()
11-17 19:24:00.172 25935  8421 E KDE/Device: org.kde.kdeconnect.Helpers.SMSHelper$MessageAccessException: Error getting messages from content://mms-sms/conversations?simple=true . Available columns were: [_id, date, message_count, recipient_ids, snippet, snippet_cs, read, archived, type, error, has_attachment, attachment_info, notification]
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Helpers.SMSHelper.getMessages(SMSHelper.java:297)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Helpers.SMSHelper.getConversations(SMSHelper.java:335)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Plugins.SMSPlugin.SMSPlugin.handleRequestConversations(SMSPlugin.java:393)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Plugins.SMSPlugin.SMSPlugin.onPacketReceived(SMSPlugin.java:329)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Device.onPacketReceived(Device.java:569)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Backends.BaseLink.packageReceived(BaseLink.java:84)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.receivedNetworkPacket(LanLink.java:255)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.LanLink.lambda$reset$0$LanLink(LanLink.java:109)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Backends.LanBackend.-$$Lambda$LanLink$TabvaCXA5qL_bcJDmIELWusNThw.run(lambda)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at java.lang.Thread.run(Thread.java:761)
11-17 19:24:00.172 25935  8421 E KDE/Device: Caused by: android.database.sqlite.SQLiteException: no such column: address (code 1): , while compiling: SELECT address, thread_id, sub_id, text_only, read, date, body, _id, type, msg_box FROM threads ORDER BY  date DESC
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:179)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at android.database.DatabaseUtils.readExceptionFromParcel(DatabaseUtils.java:135)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at android.content.ContentProviderProxy.query(ContentProviderNative.java:421)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at android.content.ContentResolver.query(ContentResolver.java:534)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at android.content.ContentResolver.query(ContentResolver.java:475)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	at org.kde.kdeconnect.Helpers.SMSHelper.getMessages(SMSHelper.java:229)
11-17 19:24:00.172 25935  8421 E KDE/Device: 	... 9 more
`

Thank you

**Comment by ferdnyc** [comment]
> I don't see how to sign up on the kde's gitlab to fill up an issue. Have you an account ?

The KDE GitLab issue tracker is for internal use, you should file any issues in [KDE Bugzilla](https://bugs.kde.org/enter_bug.cgi?product=kdeconnect&Bugzilla_restrictlogin=on&GoAheadAndLogIn=Log%20in).

**Comment by wargreen** [comment]
Bug filled :
https://bugs.kde.org/show_bug.cgi?id=414306

**Comment by andyholmes** [comment]
Should be fixed in [aa735c22b2934258b6075e79e202d3f7cb8f8f66](https://invent.kde.org/kde/kdeconnect-android/commit/aa735c22b2934258b6075e79e202d3f7cb8f8f66).

**Comment by krzysdz** [comment]
> 
> 
> Should be fixed in [aa735c22b2934258b6075e79e202d3f7cb8f8f66](https://invent.kde.org/kde/kdeconnect-android/commit/aa735c22b2934258b6075e79e202d3f7cb8f8f66).

Unless LineageOS behaves like stock Samsung (I hope it doesn't), because then there will be a `NullPointerException` thrown (instead of a `SQLiteException`).

**Comment by andyholmes** [comment]
Closing this since the particular upstream bug has been closed and any other would fall under #320.
